src/sentry/integrations/opsgenie/actions/notification.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+import logging
+
+from sentry.integrations.opsgenie.actions import OpsgenieNotifyTeamForm
+from sentry.integrations.opsgenie.client import OpsgenieClient
+from sentry.rules.actions import IntegrationEventAction
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.services.hybrid_cloud.integration.model import RpcOrganizationIntegration
+from sentry.shared_integrations.client.proxy import infer_org_integration
+from sentry.shared_integrations.exceptions import ApiError
+
+logger = logging.getLogger("sentry.integrations.opsgenie")
+
+
+class OpsgenieNotifyTeamAction(IntegrationEventAction):
+    id = "sentry.integrations.opsgenie.notify_action.OpsgenieNotifyTeamAction"
+    form_cls = OpsgenieNotifyTeamForm
+    label = "Send a notification to Opsgenie account {account} and team {team}"
+    prompt = "Send an Opsgenie notification"
+    provider = "opsgenie"
+    integration_key = "account"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.form_fields = {
+            "account": {
+                "type": "choice",
+                "choices": [(i.id, i.name) for i in self.get_integrations()],
+            },
+            "team": {"type": "choice", "choices": self.get_teams()},
+        }
+
+    def _get_team(self, org_integration: RpcOrganizationIntegration | None):
+        if not org_integration:
+            return None
+        teams = org_integration.config.get("team_table")
+        if not teams:
+            return None
+        team_id = self.get_option("team")
+        for team in teams:
+            if team["id"] == team_id:
+                return team
+        return None
+
+    def after(self, event, state):
+        integration = self.get_integration()
+        if not integration:
+            logger.exception("Integration removed, but the rule still refers to it")
+            return
+
+        org_integration = self.get_organization_integration()
+        if not org_integration:
+            org_integration_id = infer_org_integration(
+                integration_id=integration.id, ctx_logger=logger
+            )
+            org_integration = integration_service.get_organization_integration(
+                integration_id=integration.id, organization_id=org_integration_id
+            )
```

Comment:
We shouldn't use the `infer_org_integration` here, since it's mainly a hybrid cloud utility method. We only use it for hybrid cloud since we want integrations to log the associated organizations when making requests, but historically they haven't always done that. In those cases, we just use the first valid organization and call it the 'requestor' and log that (hence passing in `ctx_logger`)

In this use case, if we don't find that this organization is associated with the integration, we should not fire the alert, rather than fire it on behalf of a different organization.